proto/trailers.proto

```diff
@@ -0,0 +1,34 @@
+import "google/protobuf/descriptor.proto";
+
+option go_package = "github.com/momentohq/client-sdk-go;client_sdk_go";
+option java_package = "momento.trailers";
+option csharp_namespace = "Momento.Protos.Trailers";
+
+package trailers;
+
+// Found on the `retry_disposition` trailer for error responses, when the value is known by the service
+enum RetryDisposition {
+     // This rpc is safe to retry, even if it is non-idempotent. It was not executed by the server.
+    retryable = 0;
+    // This rpc may be safe to retry, but it may have been applied.
+    // Non-idempotent commands should not be retried, unless configured explicitly.
+    // Idempotent commands should be considered eligible for retry.
+    possibly_retryable = 1;
+}
+
+// Found on the `err` trailer for error responses, when the error is sent by the server and it has a relevant value to set
+enum Err {
+    // This is to indicate the error is coming from Momento, and not tonic or other middleware
+    general_momento_error = 0;
+    // The server may or may not have tried to process this command, but it was unable
+    // to complete it either way. A resource was exhausted which prevented the request
+    // from completing.
+    server_is_busy = 1;
+    // Indicates that the stored type for the key supplied isn't compatiable with the 
+    // requested operation
+    invalid_type = 2;
+    // Indicates the item doesn't exist for the given key
+    item_not_found = 3;
+    // Indicates the store doesn't exist
+    store_not_found = 4;
```

Comment:
Will these trailers be used across many services, or are they be specific to the storage client? I ask since the enum variant here is store-specific but the file is called (non-specific) `trailers`.

Reply:
These trailers will be used in many services, we are just starting this pattern with the addition of the new store service.

Reply:
So we would be adding service-specific (e.g. store_not_found) errors to a shared proto that could be used by other services besides the storage service?

Not that I have an overwhelming objection to that but just want to make sure we are all clear/aligned.